service-base/src/main/java/org/eclipse/hono/service/command/CommandSubscription.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.service.command;
+
+import org.eclipse.hono.service.auth.device.Device;
+import org.eclipse.hono.util.CommandConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+/**
+ * The MQTT subscription of devices, to get commands.
+ *
+ * <p>
+ * Format of subscription need to be: {@code control|c/+|TENANT/+|DEVICE_ID/req|q/#} - e.g.:
+ * </p>
+ * <ol>
+ * <li>{@code control/+/+/req/#} - authenticated device and verbose format</li>
+ * <li>{@code c/+/+/q/#} - authenticated device with short format</li>
+ * <li>{@code control/DEFAULT_TENANT/4711/req/#} unauthenticated device with verbose format</li>
+ * </ol>
+ */
+public class CommandSubscription {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CommandSubscription.class);
+
+    private String endpoint;
+    private String req;
+    private String tenant;
+    private String deviceId;
+    private boolean isAutenticated;
+
+    private CommandSubscription(final String topic) {
+        Objects.requireNonNull(topic);
+        final String[] parts = topic.split("\\/");
+        if (parts.length != 5 || !"#".equals(parts[4])) {
```

Comment:
can't we use `ResourceIdentifier` for this?

Reply:
You mean for the complete subscription part? I think it is very specific.

Reply:
What is wrong with the actual implementation? It points you better to the specific error as a regular expression.